bobtimus/src/lib.rs

```diff
@@ -84,7 +96,62 @@ impl<R, RS> Bobtimus<R, RS> {
             change_address,
         );
 
-        let protocol_state = protocol_state.interpret(&mut self.rng, payload.protocol_msg)?;
+        let alice_inputs = payload
+            .alice_inputs
+            .iter()
+            .copied()
+            .map(
+                |AliceInput {
+                     outpoint,
+                     blinding_key,
+                 }| {
+                    let client = self.elementsd.clone();
+                    async move {
+                        let transaction = client.get_raw_transaction(outpoint.txid).await.with_context(|| format!("failed to fetch transaction {}", outpoint.txid))?;
+
+                        let txin = TxIn {
+                            previous_output: outpoint,
+                            is_pegin: false,
+                            has_issuance: false,
+                            script_sig: Default::default(),
+                            sequence: 0,
+                            asset_issuance: Default::default(),
+                            witness: Default::default(),
+                        };
+                        let txin_as_txout = transaction
+                            .output
+                            .get(outpoint.vout as usize)
+                            .with_context(|| {
+                                format!(
+                                    "vout index {} is not valid for transaction {}",
+                                    outpoint.vout, outpoint.txid
+                                )
+                            })?
+                            .clone();
+
+                        Result::<_, anyhow::Error>::Ok((txin, txin_as_txout, blinding_key))
+                    }
+                },
+            )
+            .collect::<FuturesUnordered<_>>()
+            .try_collect::<Vec<_>>()
+            .await?;
```

Comment:
I freakin' love how concise this is in Rust!

This executes all futures in parallel and returns us a `Result<Vec<_>, _>`!